Add instructions to install istio 1.6 (#98)
mulesoft/docs-service-mesh: modules/ROOT/pages/prepare-to-install-service-mesh.adoc

|===

[[install_Istio]]
== Download, Install, and Configure Istio

You can install Istio 1.6.x, 1.5.x, or 1.4.x in your environment to support Anypoint Service Mesh. However, the Istio configuration for Anypoint Service Mesh differs based on the Istio version that you installed.

> === Task Prerequisites

Reviewer: ```suggestion
=== Prerequisites
```

Reviewer 2: Done! Thanks!

[[install_Istio]]

> == Installing Istio

Reviewer: ```suggestion
== Install Istio
```

Reviewer 2: Done! Thanks!

> You can install Istio 1.6.x, 1.5.x, or 1.4.x in your environment to support Anypoint Service Mesh. However, the Istio configuration for Anypoint Service Mesh differs based on the Istio version that you installed.

Reviewer: Are we documenting installation or configuration?
Seems like the latter:
```suggestion
Although you can install Istio 1.6.x, 1.5.x, or 1.4.x in your environment to support Anypoint Service Mesh, the Istio configuration for Anypoint Service Mesh depends on the Istio version that you installed.
```

Reviewer: Any prerequisites that apply to all three installations (=== Prerequisites)?

Reviewer 2: None! Done! Thanks!

> === Install Istio 1.6.x

Reviewer: ```suggestion
=== Install and Configure Istio 1.6.x
```

Reviewer 2: Done! Thanks!

> Anypoint Service Mesh supports Istio versions `1.6.0` through `1.6.8`. To install and configure Istio 1.6.x for Anypoint Service Mesh:

Reviewer: ```suggestion
Anypoint Service Mesh supports Istio 1.6.x versions 1.6.0 through 1.6.8. To install and configure Istio 1.6.x for Anypoint Service Mesh:
```

Reviewer: Consider whether there are prereqs for installing 1.6.x. 
If so, document them following the first sentence above (==== Prerequisites) and then use the subtitle ==== Steps to transition from prereqs to steps.

Reviewer 2: Done! Thanks!

> . Download Istio using the https://istio.io/docs/setup/[Istio documentation].

Reviewer: Downloading is not part of installing (and configuring). 
This looks like it should be a prerequisite, not the first step of installing.

Reviewer 2: Done! Thanks!

> . Install Istio with the following configurations:

Reviewer: How does one "install with configurations"? Doesn't one first install and then configure?
1. Download
2. Install
3. Enable flag
4. Enable telemetry
5. Install manifest? What is a manifest? It sounds like it is separate from Istio, which has already been installed. What is a full manifest vs. a not full manifest?

The topic titles here are incorrect if the task includes three tasks:
1. Download
2. Install
3. Configure

Review to ensure that the language used throughout this topic is precise.

Reviewer 2: Done! Thanks!

.. Enable the policy control flag:
+
[source,text,linenums]
----
values:
  meshConfig:
    disablePolicyChecks: false
  components:
    policy:
      enabled: true    
----
.. Enable telemetry:
+
[source,text,linenums]
----
values:
  telemetry:
    v1:
      enabled: true
    v2:
      enabled: false
components:
  citadel:
    enabled: true
  telemetry:
    enabled: true
----

> The following example illustrates a full manifest: 

Reviewer: What is this? How does it relate to:
1. Installing
2. Enabling flag
3. Enabling telemetry

It's position indicates that it is an example of enabling telemetry.

Reviewer: ```suggestion
With the policy control flag and telemetry enabled, the configuration looks like this:
```
Where does "manifest" fit in?

Reviewer 2: Done! Thanks!

+
[source,text,linenums]
----
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  profile: default
  components:
    policy:
      enabled: true
    telemetry:
      enabled: true
    ingressGateways:
    - enabled: false
  meshConfig:
    disablePolicyChecks: false
  values:
    telemetry:
      v1:
        enabled: true
      v2:
        enabled: false
----  

> When using the full manifest, ensure that you choose the correct profile. In the example, the `default` profile is used. 

Reviewer: So...doesn't configuration begin with specifying the correct profile? Profile is the first component of the install process, so it should be described first, shouldn't it?
1. Install
2. Specify profile
3. Enable policy control flag
4. Enable telemetry

Reviewer 2: Left a question for the engineers. Done! Thanks!

> To install the manifest, run:

Reviewer: This is worded as if it is a whole separate task, if installing the manifest is not part of installing Istio.
Clarify.
If this is a step in the install task, it should be numbered and placed in its chronological location.

Reviewer 2: Left a comment for the engineers. Done! Thanks!

+
`istioctl manifest apply -f <manifest-file.yaml>`

=== Install Istio 1.5.x

> Anypoint Service Mesh supports Istio versions `1.5.0` through `1.5.9`. To install and configure Istio 1.5.x for Anypoint Service Mesh:

Reviewer: ditto same edits made to similar sentence for 1.6.x

Reviewer 2: Nope the supported versions are as have been documented. Done! Thanks!

=== Install Istio 1.4.x

> Anypoint Service Mesh supports Istio versions `1.4.0` through `1.4.10`. To install and configure Istio 1.4.x for Anypoint Service Mesh:

Reviewer: ditto

Reviewer 2: Done! Thanks!
